client/src/extension.ts

```diff
@@ -51,6 +51,27 @@ import { QUICKFIX_COMMAND, CoqWarningQuickFix } from './QuickFixProvider';
 let client: Client;
 
 export function activate(context: ExtensionContext) {
+
+    // Function to check for Coq files present in the workspace
+    function checkCoqFilesPresent() {
+      if (workspace.workspaceFolders) {
+        // Search for Coq files (*.v) in the current workspace
+        workspace.findFiles('**/*.v').then(files => {
+          const hasCoqFiles = files.length > 0;
+          commands.executeCommand('setContext', 'coqFilesPresent', hasCoqFiles);
+        });
+      } else {
+        commands.executeCommand('setContext', 'coqFilesPresent', false);
+      }
+    }
+
+    // Set initially on extension activation
+    checkCoqFilesPresent();
```

Comment:
If you use the activation event this simply can be replaced by a 
```typescript
commands.executeCommand('setContext', 'coqFilesPresent', true);
```
Don't forget to "deset" it in the deactivate function, i.e. add
```typescript
commands.executeCommand('setContext', 'coqFilesPresent', undefined);
```
in the deactivate function

Reply:
I suppose on top of that, it could activate if other files are present such as a _CoqProject file ?
So maybe call the context ```'inCoqProject'``` ?